Data/ByteString/Lazy.hs

```diff
@@ -1272,7 +1278,7 @@ hGetNonBlocking = hGetNonBlockingN defaultChunkSize
 
 -- | Read an entire file /lazily/ into a 'ByteString'.
 -- The Handle will be held open until EOF is encountered.
---
+-- Note that this function's implementation relies on 'hGetContents'
```

Comment:
The importance of this remark may be unclear for readers: `Data.ByteString.Lazy` is a long module and the note at the beginning would not be visible at this point.